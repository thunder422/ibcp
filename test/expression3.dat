##############################################
#  Expression Test #3: Parenthetical Tokens  #
##############################################

MID$(A$+B$,(C+5)*D,4)+" Test"
int(Arr(A,B*(C+2))+(D))
Array(INT(B(X, Y * (-I + J), FNZ(I))), VAL(NUM$))

# begin of error tests
INT(A+B
B,F+G
Arr(B,,C)
MID$(,,C)
MID$(A,,C)
MID$(A$,,C)
MID$(A$,B$,C)
MID$(A$,B,,C)
MID$(A$,B,C$)
MID$(A$,B,C,)
