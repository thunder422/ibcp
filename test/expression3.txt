
Input: MID$(A$+B$,(C+5)*D,4)+" Test"
Output: A$ B$ +$[A$,B$] C 5 +%2 D * CvtInt 4 MID3$( " Test" +$1[" Test"] 

Input: int(Arr(A,B*(C+2))+(D))
Output: A<ref> B C 2 +%2 * Arr([A<ref>,*] D')' + INT( 

Input: Array(INT(B(X, Y * (-I + J), FNZ(I))), VAL(NUM$))
Output: X<ref> Y I Neg J + * I<ref> FNZ([I<ref>] B([X<ref>,*,FNZ(] INT( NUM$ VAL([NUM$] Array(<ref>[INT(,VAL(] 

Input: INT(A+B
              ^-- expected operator or closing parentheses


Input: B,F+G
        ^-- expected operator or end-of-statement


Input: Arr(B,,C)
             ^-- expected expression


Input: MID$(,,C)
            ^-- expected string expression


Input: MID$(A,,C)
            ^-- expected string expression


Input: MID$(A$,,C)
               ^-- expected numeric expression


Input: MID$(A$,B$,C)
               ^^-- expected numeric expression


Input: MID$(A$,B,,C)
                 ^-- expected numeric expression


Input: MID$(A$,B,C$)
                 ^^-- expected numeric expression


Input: MID$(A$,B,C,)
                  ^-- expected operator or closing parentheses


