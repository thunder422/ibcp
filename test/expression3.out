
Input: MID$(A$+B$,(C+5)*D,4)+" Test"
Output: MID$(A$ + B$, (C + 5) * D, 4) + " Test" 

Input: int(Arr(A,B*(C+2))+(D))
Output: INT(Arr(A, B * (C + 2)) + (D)) 

Input: Array(INT(B(X, Y * (-I + J), FNZ(I))), VAL(NUM$))
Output: Array(INT(B(X, Y * (-I + J), FNZ(I))), VAL(NUM$)) 

Input: INT(A+B
              ^-- expected operator or closing parentheses

Input: B,F+G
        ^-- expected operator or end-of-statement

Input: Arr(B,,C)
             ^-- expected numeric expression

Input: MID$(,,C)
            ^-- expected string expression

Input: MID$(A,,C)
            ^-- expected string expression

Input: MID$(A$,,C)
               ^-- expected numeric expression

Input: MID$(A$,B$,C)
               ^^-- expected numeric expression

Input: MID$(A$,B,,C)
                 ^-- expected numeric expression

Input: MID$(A$,B,C$)
                 ^^-- expected numeric expression

Input: MID$(A$,B,C,)
                  ^-- expected operator or closing parentheses

